packages/amplication-data-service-generator/src/util/ast.ts

```diff
@@ -769,3 +769,70 @@ export function createGenericArray(
     builders.tsTypeParameterInstantiation([itemType])
   );
 }
+
+/**
+ * Returns the first decorator with a specific name from the given AST
+ * @param ast the AST to return the decorator from
+ */
+export function findFirstDecoratorByName(
+  node: ASTNode,
+  decoratorName: string
+): namedTypes.Decorator {
+  let decorator: namedTypes.ClassDeclaration | null = null;
+  recast.visit(node, {
+    visitDecorator(path) {
+      const callee = path.get("expression", "callee");
+      if (callee.value && callee.value.name === decoratorName) {
+        decorator = path.value;
+        return false;
+      }
+      return this.traverse(path);
+    },
+    // Recast has a bug of traversing class decorators
+    // This method fixes it
+    visitClassDeclaration(path) {
+      const childPath = path.get("decorators");
+      if (childPath.value) {
+        this.traverse(childPath);
+      }
+      return this.traverse(path);
+    },
+    // Recast has a bug of traversing class property decorators
+    // This method fixes it
+    visitClassProperty(path) {
+      const childPath = path.get("decorators");
+      if (childPath.value) {
+        this.traverse(childPath);
+      }
+      this.traverse(path);
```

Comment:
don't need to return here?

Reply:
The reason we implement visitClassProperty and visitClassDeclaration is to make the visit method traverse into decorators - there is a bug in the package.
So the idea is just to check if there are any decorators to the class or property and traverse into it